### PR TITLE
[SPARK-47497][SQL][FOLLOWUP] Add a UT for `nested structure` for the function `to_csv`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to add a UT for `nested structure` for the function `to_csv`.
FollowUp: https://github.com/apache/spark/pull/45657

### Why are the changes needed?
Add a UT, improve test coverage.
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Manually test.
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @LuciferYang

**Review by LuciferYang — APPROVED** [review]
+1, LGTM (pending test)
also cc @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala [hunk 1]
```diff
@@ -768,4 +768,32 @@ class CsvFunctionsSuite extends QueryTest with SharedSparkSession {
...
+      StructField("name", StringType),
+      StructField("scores", ArrayType(MapType(StringType, LongType)))))
+    val schema = StructType(Seq(
+      StructField("key", LongType),
+      StructField("value", valueSchema)))
+
+    val df = spark.createDataFrame(rows, schema)
+    val actual1 = df.select(to_csv($"value"))
+    checkAnswer(actual1, Row("2,Alice," +
+      "\"[{math -> 100, english -> 200, science ->}, " +
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala:790]: Are we really going to settle this down as a formal behaviour? Is this upported in Spark Connect with Scala client?

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala:790]: Yes, this be supported in Spark Connect with Scala client, double-check as follows:
image

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala (hunk 1 above)
Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala:790]: What about Spark Connect with Python?

Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala:790]: Ah, okay it's fixed in https://github.com/apache/spark/pull/45657

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**